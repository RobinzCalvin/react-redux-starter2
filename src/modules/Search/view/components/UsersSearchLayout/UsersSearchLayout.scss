.users-search-layout {
  &__search-form {
    margin: 0 auto 1rem;
    width: 30rem;

    @media (max-width: 490px) {
      width: 95%;
    }
  }

  &__pagination {
    margin-top: 1.5rem;
  }
}
