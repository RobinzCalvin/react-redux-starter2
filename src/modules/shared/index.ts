export { default as Layout } from './Layout/Layout';
export { default as withLayout } from './withLayout';
