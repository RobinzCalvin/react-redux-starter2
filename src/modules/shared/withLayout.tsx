import React from 'react';

import Layout from './Layout/Layout';

function withLayout(title: string) {
  return (Component: React.ComponentType) => () => (
    <Layout title={title}>
      <Component />
    </Layout>
  );
}

export default withLayout;
