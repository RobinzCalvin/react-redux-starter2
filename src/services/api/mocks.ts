import { ICakePreview } from 'shared/types/models';

// tslint:disable:max-line-length
export const cakesPreview: ICakePreview[] =  [
  {
    name: 'Vanilla Almond',
    description: 'Light and sweet but moist and rich in flavor, our Vanilla Almond cake is filled and frosted with our house vanilla buttercream. Vanilla Almond Amycakes are decorated with sparkly clear crystal sprinkles, pearl sprinkles and a ring of buttercream rosettes',
    imageURL: '',
  },
  {
    name: 'Golden Vanilla',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house vanilla buttercream. Golden Vanilla Amycakes are decorated with ascending gold sprinkles, white ganache drizzle and white chocolate curls and flakes',
    imageURL: '',
  },
  {
    name: 'Golden Chocolate',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house chocolate buttercream. Golden Chocolate Amycakes are decorated with ascending gold sprinkles, chocolate ganache drizzle and dark chocolate curls and flakes.',
    imageURL: '',
  },
  {
    name: 'Confetti',
    description: 'Moist Vanilla Almond cake studded with confetti sprinkles in the batter. Confetti Amycakes are frosted with vanilla buttercream and decorated with colorful sprinkles.',
    imageURL: '',
  },
  {
    name: 'Zebra',
    description: 'Alternating layers of vanilla almond and chocolate cakes, filled and frosted with our house vanilla buttercream.  Zebra Amycakes are decorated with chocolate ganache drizzle and white and chocolate curls. ',
    imageURL: '',
  },
];
