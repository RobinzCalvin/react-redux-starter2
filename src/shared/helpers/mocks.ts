import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export function createContainerMock() {
  return function ContainerMock() {
    return null;
  } as any;
}

export function getProfileMock(): IProfile {
  return {
    age: 100,
    nickname: 'Nickname',
    name: 'Name name',
    bio: 'profile bio',
    avatarURL: 'https://the-url.com',
  };
}

export function getGithubUserMock(): IGithubUser {
  return {
    id: 1,
    username: 'username',
    avatarURL: 'http://the-url.com',
    htmlURL: 'http://the-url.com',
  };
}

export function getDetailedGithubUserMock(): IDetailedGithubUser {
  return {
    ...getGithubUserMock(),
    followersNumber: 12,
    followingNumber: 12,
    reposNumber: 12,
    realName: 'The real name',
    location: 'Nowhere',
  };
}

export function getRepositoryMock(): IRepository {
  return {
    id: 1,
    openIssuesNumber: 1,
    starsNumber: 1,
    forksNumber: 1,
    description: 'description',
    name: 'name',
    htmlURL: 'http://the-url.com',
    language: 'en',
    updatedAt: '05/05/2005',
    owner: getGithubUserMock(),
  };
}
// TODO: моки?
