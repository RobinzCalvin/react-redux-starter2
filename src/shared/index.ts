const testContext = (require as any).context('./', true, /^.*\/(?!.*test).*\.tsx?$/);
testContext.keys().forEach(testContext);
