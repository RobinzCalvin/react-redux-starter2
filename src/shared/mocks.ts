import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export const Container: any = () => null;

export const profile: IProfile = {
  age: 100,
  nickname: 'Nickname',
  name: 'Name name',
  bio: 'profile bio',
  avatarURL: 'https://the-url.com',
};

export const githubUser: IGithubUser = {
  id: 1,
  username: 'username',
  avatarURL: 'http://the-url.com',
  htmlURL: 'http://the-url.com',
};

export const detailedGithubUser: IDetailedGithubUser = {
  ...githubUser,
  followersNumber: 12,
  followingNumber: 12,
  reposNumber: 12,
  realName: 'The real name',
  location: 'Nowhere',
};

export const repository: IRepository = {
  id: 1,
  openIssuesNumber: 1,
  starsNumber: 1,
  forksNumber: 1,
  description: 'description',
  name: 'name',
  htmlURL: 'http://the-url.com',
  language: 'en',
  updatedAt: '05/05/2005',
  owner: githubUser,
};
