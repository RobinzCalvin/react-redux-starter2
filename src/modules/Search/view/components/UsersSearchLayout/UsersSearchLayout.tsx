import React, { useState } from 'react';
import block from 'bem-cn';

import * as features from 'features';
import featureConnect from 'core/FeatureConnector';

import Layout from 'modules/shared/Layout/Layout';
import './UsersSearchLayout.scss';

interface IFeatureProps {
  usersSearchFeatureEntry: features.usersSearch.Entry;
}

type IProps = IFeatureProps;

const b = block('users-search-layout');

function UsersSearchLayout(props: IProps) {
  const [lastSubmittedFormState, setLastSubmittedFormState] =
    useState<features.usersSearch.namespace.IUsersSearchFormFields | null>(null);

  const { usersSearchFeatureEntry: { containers } } = props;
  const { UsersSearchForm, UsersSearchResults } = containers;

  return (
    <Layout title="GitHub users search">
      <div className={b()}>
        <div className={b('search-form')}>
          <UsersSearchForm onSubmit={setLastSubmittedFormState} />
        </div>
        {lastSubmittedFormState && <UsersSearchResults searchOptions={lastSubmittedFormState} />}
      </div>
    </Layout>
  );
}

export { UsersSearchLayout, IProps as IUsersSearchLayoutProps  };
export default featureConnect({
  usersSearchFeatureEntry: features.usersSearch.loadEntry,
})(UsersSearchLayout);
