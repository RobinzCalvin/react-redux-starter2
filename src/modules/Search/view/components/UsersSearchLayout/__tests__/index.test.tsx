import React from 'react';
import { render } from 'enzyme';

import { makeMockEntry } from 'shared/mocks';

import { UsersSearchLayout, IUsersSearchLayoutProps } from '../UsersSearchLayout';

const props: IUsersSearchLayoutProps = {
  usersSearchFeatureEntry: makeMockEntry(),
};

describe('UsersSearchLayout component', () => {
  it('should match snapshot', () => {
    const component = render(<UsersSearchLayout {...props} />);
    expect(component).toMatchSnapshot();
  });
});
