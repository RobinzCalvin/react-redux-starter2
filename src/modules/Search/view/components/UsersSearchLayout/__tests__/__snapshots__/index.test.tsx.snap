// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersSearchLayout component should match snapshot 1`] = `null`;
